src/core/instance/render-helpers/render-list.js

```diff
@@ -25,11 +25,27 @@ export function renderList (
       ret[i] = render(i + 1, i)
     }
   } else if (isObject(val)) {
-    keys = Object.keys(val)
-    ret = new Array(keys.length)
-    for (i = 0, l = keys.length; i < l; i++) {
-      key = keys[i]
-      ret[i] = render(val[key], key, i)
+    if (hasSymbol && val[Symbol.iterator]) {
+      const iterator:Iterator<any> = val[Symbol.iterator]()
+      ret = []
+      i = 0
+      let next
+      do {
+        next = iterator.next()
+        if (!next.done) {
+          key = i
+          ret.push(render(next.value, key, i))
```

Comment:
Why not just `ret.push(render(next.value, ret.length))`?

Reply:
good point I can update this